### PR TITLE
updated template use

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@mfojtik @deads2k PTAL

Currently this is spitting out quite a lot of:
```
[INFO] Waiting for command to finish: '[[ "$(oc get endpoints docker-registry --output-version=v1 -t "{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}" --config=${ADMIN_KUBECONFIG} || echo "0")" != "0" ]]'...
Flag shorthand -t has been deprecated, please use --template instead
Error from server: endpoints "docker-registry" not found
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
Flag shorthand -t has been deprecated, please use --template instead
```

### CODE REVIEW

**Comment by deads2k** [comment]
Search for all of them and fix them.

**Comment by stevekuznetsov** [comment]
@deads2k done

**Comment by mfojtik** [comment]
@stevekuznetsov I think you missed one in extended/core.sh

**Comment by kargakis** [comment]
While on this can you update all the uses of `--create-external-load-balancer` to `--type="LoadBalancer"` too?

**Comment by deads2k** [comment]
@kargakis since you've looked into this, can you review?

**Comment by kargakis** [comment]
> @kargakis since you've looked into this, can you review?

Sure. We have to run extended tests for this to catch any warnings. [test][extended:core]

**Comment by stevekuznetsov** [comment]
@liggitt is `TestEventf` still flaking on Travis?

@kargakis How would this cause 13 extended tests to fail?

**Comment by kargakis** [comment]
> @kargakis How would this cause 13 extended tests to fail?

Try to rebase on top of the current master. You should see 8 of them failing.

**Comment by stevekuznetsov** [comment]
@kargakis rebased, [test]

**Comment by stevekuznetsov** [comment]
I have no idea what happened during extended tests but I'm fairly certain it has nothing to do with my changes ...

**Comment by deads2k** [comment]
re[test]

**Comment by stevekuznetsov** [comment]
This failure remains unintelligible to me:
```
package github.com/openshift/origin/pkg/cmd/cli
	imports github.com/openshift/origin/pkg/cmd/cli/cmd
	imports github.com/openshift/origin/pkg/cmd/cli/describe
	imports github.com/gonum/graph/internal: use of internal package not allowed
```

**Comment by liggitt** [comment]
it's a go1.5 error... they changed package semantics... see the list email

**Comment by stevekuznetsov** [comment]
@kargakis Updated the load-balancer flag.

File: pkg/cmd/cli/cmd/expose.go [hunk 1]
```diff
@@ -179,8 +179,8 @@ func validateFlags(cmd *cobra.Command, generator string) error {
 		if cmdutil.GetFlagInt(cmd, "port") != -1 {
 			invalidFlags = append(invalidFlags, "--port")
 		}
-		if cmdutil.GetFlagBool(cmd, "create-external-load-balancer") {
-			invalidFlags = append(invalidFlags, "--create-external-load-balancer")
+		if cmdutil.GetFlagBool(cmd, "type") {
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/expose.go]: Do this in a separate pull.  I don't think this what we want to do.  If we have an old flag and a new flag with the same meaning, alias one with the other.  Don't clutter this pull with it though.

File: test/cmd/basicresources.sh [hunk 2]
```diff
@@ -56,7 +56,7 @@ echo "routes: ok"
 
 # Expose service as a route
 oc create -f test/integration/fixtures/test-service.json
-[ ! "$(oc expose service frontend --create-external-load-balancer)" ]
+[ ! "$(oc expose service frontend --type="LoadBalancer")" ]
```
Comment (deads2k) [inline test/cmd/basicresources.sh]: How can a bool flag take a non-bool arg?

**Comment by deads2k** [comment]
> We may need to push this through faster, it's causing test-cmd to fail on master right now:

That would be really strange since the merge queue tests it before committing.  I'd bet on you forgetting to rebuild or having a mismatch like that locally.

File: test/cmd/basicresources.sh (hunk 2 above)
Comment (kargakis) [inline test/cmd/basicresources.sh]: `--type` is string (https://github.com/kubernetes/kubernetes/blob/6f4bcd5fe2fb4ebfda52b4169816e057ea73ba80/pkg/kubectl/cmd/expose.go#L76)

**Comment by stevekuznetsov** [comment]
@deads2k @kargakis Removed second commit.

**Comment by deads2k** [comment]
Looks like this covers 90%+ of cases.  [merge]

**Pull request merged**